File: SeleniumDotNet - Chrome/SeleniumDotNet/Commons/WebPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace SeleniumFramework.Commons
{
    class WebPage
    {
        protected ChromeDriver driver;

        public WebPage(ChromeDriver driver)
        {
            this.driver = driver;
        }

        protected void LoadPage(string url)
        {
            driver.Url = url;
            TestCase.Log("Entering "+url);
        }

        protected void Click(string path, string pathType)
        {

            if (pathType == "id")
            {
                IWebElement element = driver.FindElement(By.Id(path));
                element.Click();
                TestCase.Log("Clicked element with id=" + path);
            }
            else if (pathType == "name")
            {
                IWebElement element = driver.FindElement(By.Name(path));
                element.Click();
                TestCase.Log("Clicked element with name=" + path);
            }
            else if (pathType == "class")
            {
                IWebElement element = driver.FindElement(By.ClassName(path));
                element.Click();
                TestCase.Log("Clicked element with class=" + path);
            }
            else if (pathType == "xpath")
            {
                IWebElement element = driver.FindElement(By.XPath(path));
                element.Click();
                TestCase.Log("Clicked element with xpath=" + path);
            }
        }

        protected void Type(string path, string pathType, string value)
        {
            if (pathType == "id")
            {
                IWebElement element = driver.FindElementById(path);
                element.Clear();
                element.SendKeys(value);
                TestCase.Log("Entered "+value+" at element with id=" + path);
            }
            if (pathType == "name")
            {
                IWebElement element = driver.FindElementByName(path);
                element.Clear();
                element.SendKeys(value);
                TestCase.Log("Entered " + value + " at element with name=" + path);
            }
            if (pathType == "class")
            {
                IWebElement element = driver.FindElementByClassName(path);
                element.Clear();
                element.SendKeys(value);
                TestCase.Log("Entered " + value + " at element with class=" + path);
            }
            if (pathType == "xpath")
            {
                IWebElement element = driver.FindElementByXPath(path);
                element.Clear();
                element.SendKeys(value);
                TestCase.Log("Entered " + value + " at element with xpath=" + path);
            }
        }

        protected void SelectByText(string path, string pathType, string value)
        {
            if (pathType == "id")
            {
                SelectElement select = new SelectElement(driver.FindElementById(path));
                select.SelectByText(value);
                TestCase.Log("Selected " + value + " at element with id=" + path);
            }
            if (pathType == "name")
            {
                SelectElement select = new SelectElement(driver.FindElementByName(path));
                select.SelectByText(value);
                TestCase.Log("Selected " + value + " at element with name=" + path);
            }
            if (pathType == "class")
            {
                SelectElement select = new SelectElement(driver.FindElementByClassName(path));
                select.SelectByText(value);
                TestCase.Log("Selected " + value + " at element with class=" + path);
            }
            if (pathType == "xpath")
            {
                SelectElement select = new SelectElement(driver.FindElementByXPath(path));
                select.SelectByText(value);
                TestCase.Log("Selected " + value + " at element with xpath=" + path);
            }
        }
    }
}

File: JSelenium/src/jselenium/commons/WebPage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package jselenium.commons;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

/**
 *
 * @author Number Cruncher Mk.2
 */
public class WebPage {

    protected WebDriver driver;

    public WebPage(WebDriver driver) {
        this.driver = driver;
    }

    protected void LoadPage(String url) {
        driver.navigate().to(url);
        TestCase.Log("Entering " + url);
    }

    protected void Click(String path, String pathType) {

        if (pathType == "id") {
            WebElement element = driver.findElement(By.id(path));
            element.click();
            TestCase.Log("Clicked element with id=" + path);
        } else if (pathType == "name") {
            WebElement element = driver.findElement(By.name(path));
            element.click();
            TestCase.Log("Clicked element with name=" + path);
        } else if (pathType == "class") {
            WebElement element = driver.findElement(By.className(path));
            element.click();
            TestCase.Log("Clicked element with class=" + path);
        } else if (pathType == "xpath") {
            WebElement element = driver.findElement(By.xpath(path));
            element.click();
            TestCase.Log("Clicked element with xpath=" + path);
        }
    }

    protected void Type(String path, String pathType, String value) {
        if (pathType == "id") {
            WebElement element = driver.findElement(By.id(path));
            element.sendKeys(value);
            TestCase.Log("Entered " + value + " at element with id=" + path);
        }
        if (pathType == "name") {
            WebElement element = driver.findElement(By.name(path));
            element.sendKeys(value);
            TestCase.Log("Entered " + value + " at element with name=" + path);
        }
        if (pathType == "class") {
            WebElement element = driver.findElement(By.className(path));
            element.sendKeys(value);
            TestCase.Log("Entered " + value + " at element with class=" + path);
        }
        if (pathType == "xpath") {
            WebElement element = driver.findElement(By.xpath(path));
            element.sendKeys(value);
            TestCase.Log("Entered " + value + " at element with xpath=" + path);
        }
    }

    protected void SelectByText(String path, String pathType, String value) {
        if (pathType == "id") {
            Select select = new Select(driver.findElement(By.id(path)));
            select.selectByVisibleText(value);
            TestCase.Log("Selected " + value + " at element with id=" + path);
        }
        if (pathType == "name") {
            Select select = new Select(driver.findElement(By.name(path)));
            select.selectByVisibleText(value);
            TestCase.Log("Selected " + value + " at element with id=" + path);
        }
        if (pathType == "class") {
            Select select = new Select(driver.findElement(By.className(path)));
            select.selectByVisibleText(value);
            TestCase.Log("Selected " + value + " at element with id=" + path);
        }
        if (pathType == "xpath") {
            Select select = new Select(driver.findElement(By.xpath(path)));
            select.selectByVisibleText(value);
            TestCase.Log("Selected " + value + " at element with id=" + path);
        }
    }
}
